vshard/replicaset.lua

```diff
@@ -421,6 +565,26 @@ local function replica_call(replica, func, args, opts)
     assert(opts and opts.timeout)
     replica.activity_ts = fiber_clock()
     local conn = replica.conn
+    if opts.is_async then
+        -- Async call cannot yield. So, we cannot wait for the connection
+        -- to be established and validate vconnect. Immediately fail, if
+        -- something is wrong.
+        local ok, err = conn_vconnect_check_or_close(conn)
+        if not ok then
+            return false, nil, lerror.make(err)
+        end
+    end
+    local timeout, err = conn_vconnect_wait_or_close(conn, opts.timeout)
+    if not timeout then
+        return false, nil, lerror.make(err)
+    end
```

Comment:
This shouldn't be on the `is_async` path, should it? For `is_async` it would either yield or return immediately which makes it either needless or a bug.

Reply:
It's NoOp in case of async, we already checked, that it's ready. This is the reason, why all tests passses.

I changed the code in this part a little bit, hope it's ok now.